### ISSUE TITLE
Gradle - Canonical version from gradle.properties

### ISSUE THREAD

**Issue by monowai** [dependabot/dependabot-core#2046]
Similar to issue 775

I manage versions using `gradle.properties`.  I've found this to be the safest way of registering a version for plugins and classpath dependencies.

When I have the version number set directly in `build.gradle`, dependencies are correctly detected as out of date.

When set as a variable, Dependabot seems unable to resolve the version of the library I'm using and simply says
`Looks like these dependencies are no longer a dependency, so this is no longer needed.`

For example:
Resilience4J, I have 
 `gradle.properties`  -> `resilience4jVersion=1.1.0` 
 `build.gradle` -> `io.github.resilience4j:resilience4j-spring-boot2:$resilience4jVersion`

**Comment by fnobilia** [comment]
Any updates on this issue? =)

**Comment by singhbaljit** [comment]
This is the only thing preventing me from using Depandabot!

**Comment by missedone** [comment]
looks like duplicates with #1618

**Comment by monowai** [comment]
Not quite resolves but #1618 does explain how to obtain the same result through a bit of reconfiguration. 
Moving the library properties from `gradle.properties` and into `build.gradle` in the root project resolves this issue.
for example:
```
buildscript {
    ext{
        springBootVersion='2.3.8.RELEASE'
        ...
```        
So while setting the propeties in `gradle.properties` works fine for Gradle, it does not work for `dependabot` which expects them to be in `build.gradle`

[My working config for referece](https://github.com/monowai/beancounter/blob/master/build.gradle)
